include/ppx/ui_util.h

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ppx_moving_average_h
+#define ppx_moving_average_h
+
+#include <cmath>
+#include <cstdint>
+#include <type_traits>
+
+#include "ppx/log.h"
+#include "ppx/timer.h"
+
+namespace ppx {
+
+template <typename FloatT>
+struct ParallelVariance
+{
+    FloatT weight = 0.0;
+    FloatT mean   = 0.0;
+    FloatT accVar = 0.0;
+
+    static ParallelVariance Combind(const ParallelVariance& a, const ParallelVariance& b)
+    {
+        // Chan et al. online variance algorithm
+        // https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Parallel_algorithm
+        //
+        // Large weight might cause problem with mean calculation.
+        // Since we are doing it for display in UI, that's not a issue.
+        const FloatT delta  = b.mean - a.mean;
+        const FloatT weight = a.weight + b.weight;
+        return {
+            weight,
+            a.mean + delta * b.weight / weight,
+            a.accVar + b.accVar + delta * delta * a.weight * b.weight / weight,
```

Comment:
nit: doesn't change anything, but IMO it's slight more readable to have
`(delta * delta * a.weight * b.weight) / n`